lib/kafka_ex/abstract_socket.ex

```diff
@@ -0,0 +1,82 @@
+defmodule KafkaEx.AbstractSocket do
+  @moduledoc false
+
+  defmodule Socket do
```

Comment:
It doesn't seem like we need this as a nested struct module.  This could just be one module called `Socket` with the `defstruct` at the top.